examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/ntp.c

```diff
@@ -0,0 +1,301 @@
+/*TODO: This interface is currently blocking at sntp_get_datetime() and only
+ *      supports 1 concurrent request. Change last_sntp_response to use a
+ *      flexible array and we can make it non-blocking and support as many
+ *      concurrent connections as necessary
+ */
+#include "stdio.h"
+#include "string.h"
+#include "wifi_interface.h"
+#include "wifi_module.h"
+#include "wifi_globals.h"
+
+#include "ntp.h"
+
+int32_t sntp_current_time = 0;
+sntp_response* last_sntp_response = NULL;
+
+/**
+   * @brief  free the space used by an sntp_response
+   * @param  *sntp_server is a char array with the relevant URL
+   * @retval WiFi_Status_t: WiFi_MODULE_SUCCESS on success, see wifi_interface.h
+   */
+void sntp_free_response( sntp_response* r )
+{
+    if( NULL != r->response )
+    {
+        free(r->response);
+    }
+    free(r);
+}
+
+/**
+   * @brief  Builds a char array with a basic SNTP request to be sent via UDP
+   *         Packet description:
+   *           - Flags 1 byte
+   *               * Leap: 3 bits
+   *               * Version: 3 bits
+   *               * Mode: 2 bits
+   *           - Stratum 1 byte
+   *           - Polling 1 byte
+   *           - Precision 1 byte
+   *           - Root Delay 4 bytes
+   *           - Root Dispersion 4 bytes
+   *           - Reference Identifier 4 bytes
+   *           - Reference Timestamp 8 bytes
+   *           - Origin Timestamp 8 bytes
+   *           - Receive Timestamp 8 bytes
+   *           - Transmit Timestamp 8 bytes
+   * @param  request_buf is a pointer to a pre-allocated 48 byte char array
+   * @retval void
+   */
+void sntp_build_request( char* request_buf )
+{
+    char sntp_pkt[SNTP_MSG_SIZE] = {
```

Comment:
2 things here.
1. Why so cumbersome? Why don't we just declare a const char array and pass the address of this to wifi_socket_client_write? Why copying?
2. I don't know the SNTP request UDP message content at all, but looked at file xi_bsp_time_cc3200_sntp.c and there the 48 lenght message is much simpler, pls check it. Although this 2nd is just a note if this works then ok.